Metadata/ListMetadataVisitor.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ArticleBundle\Metadata;
+
+use Sulu\Bundle\AdminBundle\Metadata\ListMetadata\ListMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\ListMetadata\ListMetadataVisitorInterface;
+use Sulu\Component\Content\Compat\Structure\StructureBridge;
+use Sulu\Component\Content\Compat\StructureManagerInterface;
+
+class ListMetadataVisitor implements ListMetadataVisitorInterface
```

Comment:
mark services `@internal`

```suggestion
/**
 * @final
 * @internal
 */
class ListMetadataVisitor implements ListMetadataVisitorInterface
```